lib/trino-filesystem/src/main/java/io/trino/filesystem/memory/MemoryFileSystemCacheConfig.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.memory;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.airlift.units.MaxDataSize;
+import jakarta.validation.constraints.NotNull;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.concurrent.TimeUnit.HOURS;
+
+public class MemoryFileSystemCacheConfig
+{
+    // Runtime.getRuntime().maxMemory() is not 100% stable and may return slightly different value over JVM lifetime. We use
+    // constant so default configuration for cache size is stable.
+    @VisibleForTesting
+    static final DataSize DEFAULT_CACHE_SIZE = DataSize.succinctBytes(Math.min(
```

Comment:
If the memory is less than 4 GB -> 200 megabytes (seems a lot) otherwise use 1/20th of memory (seems a lot too)

For 64 GB machine it's 3ish GB of cache. How did you come with this number? :)

Reply:
I just tried to be less aggressive than existing caches.
E.g.
The cache in delta lake is 10% of heap per catalog io.trino.plugin.deltalake.DeltaLakeConfig#DEFAULT_DATA_FILE_CACHE_SIZE.
In hive the file listing cache is 100MB per query io.trino.plugin.hive.HiveConfig#perTransactionFileStatusCacheMaxRetainedSize and 1GB in total io.trino.plugin.hive.HiveConfig#fileStatusCacheMaxRetainedSize per catalog.
The iceberg manifest cache default is 100MB per scan with 8 scans allowed to be cached at max.

Until we refactor code to allow cache to be shared across catalogs, it's hard to come up with a good number. It seems reasonable assume that someone with 4GB should mount fewer catalogs than someone with 64GB. So 5% of 4GB and 200MB from 64GB isn't too bad.

Reply:
Maybe we could go with a static cache size of 200 MB since this is configurable anyway?

Reply:
> Maybe we could go with a static cache size of 200 MB since this is configurable anyway?

Yup. Simiar as it's done for https://github.com/trinodb/trino/blob/5dc1afe9b36e24635a1cc1680a923ef6b1f9d91b/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java#L145. 

Don't make it too complicated.

Reply:
I don't think minimum of 200MB or 5% of heap size is too complicated, will stick with it.